keps/sig-auth/20200507-external-tls-certificate-authenticator/README.md

```diff
@@ -0,0 +1,749 @@
+  [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] KEP approvers have approved the KEP status as `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG
+  Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for
+  publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to
+  mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every
+time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/ [kubernetes/enhancements]:
+https://git.k8s.io/enhancements [kubernetes/kubernetes]:
+https://git.k8s.io/kubernetes [kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+This enhancement proposes adding support for authentication via external TLS
+certificate signers, what would enable usage of Hardware Security Modules (HSMs)
+- also known as smartcards, cryptographic processors or, by a popular brand
+name, YubiKeys(tm) via the PKCS#11 standard. This enhancement allows developers
+or automation pipelines to authenticate with the Kubernetes cluster, without
+requiring access to the client key, hence improving compliance and security.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+<!-- How would you react if your laptop was stolen? Are you worried about 
+attackers performing a cold boot attack to extract your Kubernetes 
+credentials?  -->
+Are you worried about someone getting access to your Kubernetes credentials?
+(For example, by extracting private keys from your laptop using malware or by
+performing a cold boot attack.) Do you already use a YubiKey for SSH and GPG,
+and wonder why you cannot use it with kubectl? If yes, then this enhancement is
+for you!
```

Comment:
the language in KEPs usually excludes personalization (i, we, you) or advertisement. :)